### ISSUE TITLE
Error when assume(false) with no data generation

### ISSUE THREAD

**Issue by DRMacIver** [hegeldev/hegel-rust#78]
The following should raise an error (it would in Hypothesis) but doesn't:

```rust
use hegel::TestCase;
use hegel::generators;

#[hegel::test]
fn test_does_not_hang_on_assume_false(tc: TestCase) {
    tc.assume(false);
}
```

**Comment by repowazdogz-droid** [comment]
Reproduced on main (9f0e0bb). A no-draw assume(false) passes silently, but booleans() then assume(false) raises “every reachable input was filtered out by assume()” with a count of 2.

The engine already sets test_is_trivial from an empty choice sequence and uses it correctly to stop searching. The terminal check at test_runner.rs:430 also excludes it, which looks like why the error is skipped. Every other conjunct there is satisfied.

That suggests the axis is exhausted space versus gave-up search rather than drew versus didn’t draw, which may be why #78 and #76 are related but not the same.

Unless you’d prefer it folded into #76, I’d like to take #78 as a small fix plus the missing regression tests.

Separately, not proposing to solve it here: suppressing FilterTooMuch also makes a fully exhausted, entirely rejected test pass. Hypothesis raises Unsatisfiable there.
